huaweicloud-sdk-python-modelarts-dataset/modelarts/manifest.py

```diff
@@ -0,0 +1,371 @@
+    return sample_json;
+
+  def save(self, path, *args):
+    """
+    save dataset to local or OBS
+    :param path: manifest output path
+    :param args: ak,sk,endpoint
+    :return: None
+    """
+    if args.__len__() < 1:
+      for sample in self.get_sample_list():
+        with open(path, "a") as f_obj:
+          value = self.__toJSON(sample);
+          json.dump(value, f_obj)
+          f_obj.write('\n')
+    else:
+      if (args.__len__() < 3):
+        raise ValueError("please input OBS path, ak, sk and endpoint.")
+      manifest_json = []
+      for sample in self.get_sample_list():
+        value = self.__toJSON(sample);
+        manifest_json.append(json.dumps(value))
+      save(manifest_json, path, args[0], args[1], args[2])
+
+
+class Sample(object):
+  def __init__(self, source, annotations=None, usage=None, inference_loc=None, id=None):
+    self.__source = source
+    self.__usage = usage
+    self.__annotation = annotations
+    self.__inference_loc = inference_loc
+    self.__id = id
+
+  def get_source(self):
+    """
+    :return "source" attribute
+    Mandatory field
+    """
+    return self.__source
+
+  def get_id(self):
+    """
+    :return "id" attribute, one of
+    Optional field
+    """
+    return self.__id
+
+  def get_usage(self):
+    """
+    :return "usage" attribute, one of
+    Optional field
+    """
+    return self.__usage
+
+  def get_inference_loc(self):
+    """
+    :return "inference_loc" attribute, one of Optional field
+    """
+    return self.__inference_loc
+
+  def get_annotations(self):
+    """
+    :return a list of class Annotation
+    Optional field
+    """
+    return self.__annotation
+
+
+class Annotation:
+
+  def __init__(self, name=None, type=None, loc=None, property=None, confidence=None, creation_time=None,
+               annotated_by=None, annotation_format=None, hard=None):
+    self.__name = name
+    self.__type = type
+    self.__annotation_loc = loc
+    self.__property = property
+    self.__hard = hard
+    self.__confidence = confidence
+    self.__creation_time = creation_time
+    self.__annotated_by = annotated_by
+    self.__annotation_format = annotation_format
+
+  def get_type(self):
+    """
+    :return type of dataset: modelarts/image_classification, modelarts/object_detection
+    Optional field
+    """
+    return self.__type
+
+  def get_name(self):
+    """
+    :return the name of this annotation, like "cat"
+    Mandatory field if get_loc is None
+    """
+    return self.__name
+
+  def get_loc(self):
+    """
+    :return in case of object detection, this will return the annotation file,
+    otherwise return null
+    Mandatory field if get_name is None
+    """
+    return self.__annotation_loc
+
+  def get_property(self):
+    """
+    :return a KV pair list
+    Optional field
+    """
+    return self.__property
+
+  def get_hard(self):
+    """
```

Comment:
what does `hard` mean ? Make it self-explanatory ?

Reply:
hard means this sample is Hard case，which is difficult to handle.